components/brave_wallet/browser/solana_tx_state_manager_unittest.cc

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/solana_tx_state_manager.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_prefs.h"
+#include "brave/components/brave_wallet/browser/json_rpc_service.h"
+#include "brave/components/brave_wallet/browser/solana_account_meta.h"
+#include "brave/components/brave_wallet/browser/solana_instruction.h"
+#include "brave/components/brave_wallet/browser/solana_transaction.h"
+#include "brave/components/brave_wallet/browser/solana_tx_meta.h"
+#include "brave/components/brave_wallet/common/brave_wallet_types.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+class SolanaTxStateManagerUnitTest : public testing::Test {
+ public:
+  SolanaTxStateManagerUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME),
+        shared_url_loader_factory_(
+            base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+                &url_loader_factory_)) {}
+  ~SolanaTxStateManagerUnitTest() override {}
+
+ protected:
+  void SetUp() override {
+    brave_wallet::RegisterProfilePrefs(prefs_.registry());
+    json_rpc_service_.reset(
+        new JsonRpcService(shared_url_loader_factory_, GetPrefs()));
+    solana_tx_state_manager_.reset(
+        new SolanaTxStateManager(GetPrefs(), json_rpc_service_.get()));
+  }
+
+  void SetNetwork(const std::string& chain_id) {
+    base::RunLoop run_loop;
+    json_rpc_service_->SetNetwork(
+        chain_id, mojom::CoinType::SOL,
+        base::BindLambdaForTesting([&](bool success) { run_loop.Quit(); }));
```

Comment:
We can call sync SetNetwork here

Reply:
Fixed in https://github.com/brave/brave-core/pull/12481/commits/909e7342d53bd8ccc0c34025af2c8a1799a1fa0e